### PR TITLE
Check if flowcell id matches for paired samples

### PR DESCRIPTION
**Pull request by pmoris**
I noticed [this comment ](https://github.com/nf-core/sarek/blob/5cc30494a6b8e7e53be64d308b582190ca7d2585/workflows/sarek/main.nf#L946) about checking the flowcell ID for paired samples while constructing GATK read groups. I was adapting the read group code for a custom pipeline and attempted a quick fix, so I thought I'd contribute it back to sarek.

> While constructing the read group from paired fastq samples, perform a check to ensure that the id is the same for (the first reads) in fastq_1 and fastq_2. Exit out with an error otherwise and report the problematic sample and file paths.

Incidentally, while researching read groups I came across the following recommendations: https://support.sentieon.com/appnotes/read_groups/. Would it be worth updating some of the fields to match these guidelines?

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - => Only tested this manually, but happy to add a proper test if you could give me a starting point. Is there already an existing test for samplesheet validation that I can add this too? I guess I will need to add "corrupt" fastq files to the nf-core test repo?
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
    - => will do this after submitting the PR so that I can link to it.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).
    - => should I do this even for such a minor contribution?

### CODE REVIEW

File: workflows/sarek/main.nf [hunk 1]
```diff
@@ -944,11 +944,16 @@ workflow SAREK {
 // Add readgroup to meta and remove lane
 def addReadgroupToMeta(meta, files) {
     def CN = params.seq_center ? "CN:${params.seq_center}\\t" : ''
+   
+    def flowcell = flowcellLaneFromFastq(files[0])
+
+    // Check if flowcell ID matches for paired samples
```
Comment (maxulysse) [inline workflows/sarek/main.nf:950]:
```suggestion
    // Check if flowcell ID matches
```

**Comment by maxulysse** [comment]
@nf-core-bot fix linting :pray: pretty please :pray:

**Comment by maxulysse** [comment]
@pmoris I updated your PR with the latest update in this function.
No need to check for paired samples as sarek only handles paired samples

**Comment by maxulysse** [comment]
Can you update the CHANGELOG

**Comment by pmoris** [comment]
Changelog is updated!

I also fixed the conditional (by removing the `meta.single_end` check, it accidentally moved the negation to the flowcell variable, causing the check to not trigger at the right time).

Lastly, what are your thoughts on updating the `PU` field to `flowcell.lane` rather than just `lane` (as recommended here: https://support.sentieon.com/appnotes/read_groups/)?

**Comment by pmoris** [comment]
Why is the linter complaining? There is no trailing whitespace or non-multiple-of-4 padding as far as I can tell...

**Review by maxulysse — APPROVED** [review]

**Pull request merged**